apps/polls/assets/PollManagement.jsx

```diff
@@ -0,0 +1,224 @@
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  handleMoveQuestionDown: function (index) {
+    var question = this.state.questions[index]
+    var diff = {$splice: [[index, 1], [index + 1, 0, question]]}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  handleAppendQuestion: function () {
+    var newQuestion = this.getNewQuestion('')
+    var diff = {$push: [newQuestion]}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  handleDeleteQuestion: function (index) {
+    var diff = {$splice: [[index, 1]]}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  /*
+  |--------------------------------------------------------------------------
+  | Choice state related handlers
+  |--------------------------------------------------------------------------
+  */
+
+  getNextChoiceKey: function () {
+    /** Get an artifical key for non-commited choices.
+     *
+     *  Prefix to prevent collisions with real database keys;
+     */
+    var choiceKey = 'local_' + (this.state.maxChoiceKey + 1)
+    this.setState({maxChoiceKey: this.state.maxChoiceKey + 1})
+    return choiceKey
+  },
+
+  getNewChoice: function (label) {
+    var newChoice = {}
+    newChoice['label'] = label
+    newChoice['key'] = this.getNextChoiceKey()
+    return newChoice
+  },
+
+  handleUpdateChoiceLabel: function (questionIndex, choiceIndex, label) {
+    var diff = {}
+    diff[questionIndex] = {choices: {}}
+    diff[questionIndex]['choices'][choiceIndex] = {$merge: {label: label}}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  handleAppendChoice: function (questionIndex) {
+    var newChoice = this.getNewChoice('')
+    var diff = {}
+    diff[questionIndex] = {choices: {$push: [newChoice]}}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  handleDeleteChoice: function (questionIndex, choiceIndex) {
+    var diff = {}
+    diff[questionIndex] = {choices: {$splice: [[choiceIndex, 1]]}}
+
+    this.setState({
+      questions: update(this.state.questions, diff)
+    })
+  },
+
+  /*
+  |--------------------------------------------------------------------------
+  | Poll form and submit logic
+  |--------------------------------------------------------------------------
+  */
+
+  handleSubmit: function (e) {
+    e.preventDefault()
+
+    let data = {
+      questions: this.state.questions
+    }
+
+    let promise = api.poll.change(data, this.props.poll.id)
+    promise
+      .done(function (data) {
+        this.setState({
+          successMessage: django.gettext('The poll has been updated.')
+        })
+
+        setTimeout(function () {
+          this.setState({
+            successMessage: ''
+          })
+        }.bind(this), 1500)
+      }.bind(this))
```

Comment:
Consider using [fat arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Functions#Arrow_functions) to get rid of all the `.bind`s